flow/util/template_filters.py

```diff
@@ -109,6 +109,61 @@ def calc_tasks(operations, name, parallel=False, allow_mixed=False):
         )
 
 
+def _parse_memory(memory):
+    """Parse memory from the memory flag passed by a user.
+
+    A valid memory flag passed to `FlowProject.submit` should have a suffix of either
+    "g" for gigabytes, or "m" for megabytes.
+
+    Parameters
+    ----------
+    memory : str
+        Required memory to reserve per node.
+
+    Returns
+    -------
+    float
+        The parsed memory (numeric value) in gigabytes.
+    """
+    try:
+        size_type = memory[-1]
+        if size_type.lower() == "m":
+            return float(memory[0:-1]) / 1024
+        elif size_type.lower() == "g":
+            return float(memory[0:-1])
+        else:
+            raise ValueError
```

Comment:
Should this default to gigabytes? If not, we need to add a message for the error.
```suggestion
            raise ValueError("Memory amounts must be specified with a suffix "g" for gigabytes or "m" for megabytes.")
```

Reply:
The command line flag currently requires users to enter memory with units in gigabytes (`g`) or megabytes(`m`). In order to default it to `gigabytes`, we need to deprecate the current behaviour. (Right?)

Reply:
Currently values without a suffix `g` or `m` are not supported. I was proposing adding a _new_ behavior (not breaking a current behavior) that would allow a numeric value with no suffix to be interpreted as gigabytes.

If you think that providing a raw number without a suffix (and interpreting it as gigabytes) _should_ be allowed, then accept the first suggestion `return float(memory)`. If you think that providing a raw number without a suffix _should not_ be allowed, then accept the second suggestion `raise ValueError(...)` because it makes the error message clear.